docs/introduction/Sidechain-SDK-extension.rst

```diff
@@ -609,4 +609,59 @@ The Backup Storage must be present inside your node data directory before starts
 The procedure fails if just a single coin box is found inside the Backup Storage.
 
 If you own some of the restored boxes and you want to see them inside your wallet, you should add your secrets inside the config file of your node (before start the node for the first time).
-You can add your secrets inside the section Wallet.genesisSecrets by appending "00" at the beginning of your secret in case it is a PrivateKey25519, "03" if it is a VrfPrivateKey or "04" if it is a SchnorrPrivateKey.
\ No newline at end of file
+You can add your secrets inside the section Wallet.genesisSecrets by appending "00" at the beginning of your secret in case it is a PrivateKey25519, "03" if it is a VrfPrivateKey or "04" if it is a SchnorrPrivateKey.
+
+Logging
+###################
+
+The SDK logging system is based on the Log4J library.
+To fire additional log messages in your application code, just declare a log4j Logger in your class and use it:
+
+::
+
+    public class MyCustomClass {
+
+        Logger logger = LogManager.getLogger(MyCustomClass.class);
+
+        public MyCustomClass(){
+          logger.debug("This is an example  debug message inside the constructor");
+        }
+    }
+
+Note: do not add any log4j library in your application pom file, as it is already loaded as a nested dependency of the SDK. 
+
+You can rely on the default logging configuration (based on Log4J library) and change just a few parameters inside the application config file, or ovveride it completely with a custom one.
+
+Default log configuration
+-------------------------------
+By default, `a predefined log4j2.xml configuration <https://github.com/HorizenOfficial/Sidechains-SDK/blob/master/sdk/src/main/resources/log4j2.xml>`_  is used.
+It redirects all logging messages to the system console and to a filesystem log file, rotated and gzipped when it reaches 50MB size (only the latest 10 are then retained).
+
+The following dynamic parameters coming from the application conf file are used, so that you can change them by just updating the application conf file:
```

Comment:
We repeat twice "application conf", maybe we can rephrase in something like "The following dynamic parameters are taken from the application configuration file but you can change them at any time."